### PR TITLE
chore: move notification processor filters to common package

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

This change allows other upcoming notification processors to use the same filtering as the email processor does.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by drodil** [comment]
@mareklibra small improvement for future processors to use the same filtering. Should probably add support to `teams` and `slack` processors once they are merged.

**Review by mareklibra — APPROVED** [review]

**Comment by mareklibra** [comment]
@freben Can you please have a look? I would merge it but there is update to the `yarn.lock`

**Comment by drodil** [comment]
Not stale

**Comment by mareklibra** [comment]
not stale

File: plugins/notifications-common/src/filters.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+export const getProcessorFiltersFromConfig = (config: Config) => {
+  const filter: NotificationProcessorFilters = {};
+  const minSeverity = config.getOptionalString(
+    'filter.minSeverity',
+  ) as NotificationSeverity;
+  if (minSeverity) {
+    if (notificationSeverities.includes(minSeverity)) {
+      filter.minSeverity = minSeverity;
+    } else {
+      throw new Error(`Invalid minSeverity: ${minSeverity}`);
```
Comment (freben) [inline plugins/notifications-common/src/filters.ts:30]: For errors like this, I like to add a helpful `..., expected one of ${possibilities.join(', ')}` or so. Same below

**Review by freben — APPROVED** [review]

**Pull request merged**